src/containers/SearchPage/SelectNumberFromRangeFilter/SelectNumberFromRangeFilter.js

```diff
@@ -0,0 +1,169 @@
+  return !!valueRange && minValue != null && maxValue != null ? { minValue, maxValue } : null;
+};
+
+// Transforms a range object containing `minValue` and `maxValue`
+// into query parameter format, i.e. "0,1000"
+const formatToQueryParam = (range, queryParamName) => {
+  const { minValue, maxValue } = range || {};
+  // We explicitly compare `minValue` to null as comparing to '0' returns false but '0' is a valid range value.
+  const value = minValue != null && maxValue != null ? `${minValue},${maxValue}` : null;
+  return { [queryParamName]: value };
+};
+
+const SelectNumberFromRangeFilter = props => {
+  const {
+    min,
+    max,
+    step,
+    onSubmit,
+    queryParamNames,
+    initialValues,
+    label,
+    rootClassName,
+    className,
+    intl,
+    id,
+    name,
+    showAsPopup,
+    ...rest
+  } = props;
+
+  const classes = classNames(rootClassName || css.root, className);
+
+  const hasValue = value => value != null;
+
+  // Extracts initial values for form from query parameters. If a valid query parameter exists, it converts
+  // the string to an object using `convertQueryParamToObject`. If not, initializes as an empty object.
+  const parsedInitialValues =
+    initialValues && initialValues[queryParamNames]
+      ? convertQueryParamToObject(initialValues[queryParamNames])
+      : {};
+
+  const { minValue, maxValue } = parsedInitialValues || {};
+  const hasInitialValues = initialValues && hasValue(minValue) && hasValue(maxValue);
+
+  const resolvedInitialValues = {[name]: hasInitialValues ? parsedInitialValues : {}};
+
+  // Handles form submission by extracting and updating the range values from the form's current state.
+  const handleSubmit = values => {
+  const usedValue = values?.[name] ? values[name] : values;
+  const { minValue, maxValue } = usedValue || {};
+
+    return onSubmit(
+      formatToQueryParam(
+        {
+          minValue: minValue,
+          maxValue: maxValue,
+        },
+        queryParamNames
+      )
+    );
+  };
+
+  // Used to display the selected values above the filter component in the "grid" view
+  const labelSelectionForPlain = hasInitialValues ? (
+    <FormattedMessage id="RangeFilter.labelSelectedPlain" values={{ minValue, maxValue }} />
+  ) : null;
+
+  // A debounced version of the `handleSubmit` function to prevent excessive
+  // or premature submissions during rapid input changes.
+  const handleSubmitOnLive = debounce(
+    values => {
+      return handleSubmit(values);
+    },
+    400,
+    { leading: false, trailing: true }
+  );
+
+  return showAsPopup ? (
+    <FilterPopup
+      className={classes}
+      rootClassName={rootClassName}
+      label={label}
+      isSelected={hasInitialValues}
+      id={`${id}.popup`}
+      onSubmit={handleSubmit}
+      initialValues={resolvedInitialValues}
+      {...rest}
+    >
+      <Field
+        max={max}
+        min={min}
+        name={name}
+        step={step}
+        component={RangeInput}
+      />
```

Comment:
Perhaps it's best that Field and RangeInput are on the same file. 

Then this would be:
`<FieldSelectIntegerRange name={name} min={min} max={max} step={step} />`

Inside the file, you could have:

```jsx
const RangeInput = props => {
 // ...
}
const FieldSelectIntegerRange = props => {
  return (
    <Field
      max={max}
      min={min}
      name={name}
      step={step}
      component={RangeInput}
    />
  );
}
export default FieldSelectIntegerRange;
```

So, I suggest that you rename this component as **FieldSelectIntegerRange**